e2e/page-objects/newIssue.po.ts

```diff
@@ -1,64 +1,54 @@
-import { browser, by, element, ExpectedConditions } from 'protractor';
+import { expect, Page } from '@playwright/test';
 
 export class NewIssuePage {
+  readonly page: Page;
+
+  constructor(page: Page) {
+    this.page = page;
+  }
+
   async enterNewIssueTitle(title: string) {
-    return element(by.id('title')).sendKeys(title);
+    return this.page.locator('#title').fill(title);
   }
 
   async enterNewBugReportDescription(desc: string) {
-    const textArea = element(by.className('text-input-area'));
+    const textArea = this.page.locator('.text-input-area');
+    await textArea.focus();
     await textArea.clear();
-    return textArea.sendKeys(desc);
+    return textArea.fill(desc);
   }
 
   async selectSeverityDropdown() {
-    return element(by.className('severity-dropdown')).click();
+    return this.page.locator('.severity-dropdown').click();
   }
 
   async selectBugTypeDropdown() {
-    return element(by.className('bug-dropdown')).click();
+    return this.page.locator('.bug-dropdown').click();
   }
 
-  /**
-   * Selects dropdown option from Severity or Bug Type.
-   * NOTE: There is an issue where the page data isn't updated in
-   * realtime when dealing with drop-down clicks and option selection.
-   * As such stringing multiple of these actions (i.e. Select Severity Dropdown,
-   * Select Severity Option, Select BugType Dropdown, Select BugType option)
-   * can result in the test failing as the page does not update quickly and
-   * the driver receives stale data on the elements present on screen. Due
-   * to the fact that mat-option is a generic classname (which we have abstracted)
-   * the ExpectCondition would not accurately wait for the right dropdown list.
-   * But it has still been left in the function as a precautionary measure.
-   * A work-around to this is to place other actions in-between a dropdown option
-   * selection action. See 'creates new bug report' for example.
-   * @param optionNumber Position of Dropdown Option
-   * @param dropdownText Text within Dropdown Option
-   */
   async selectDropDownOption({ optionNumber, dropdownText }: { optionNumber?: number; dropdownText?: string }) {
```

Comment:
Perhaps we can define the parameters for tests like this as a separate object above like this:

```ts
type DropdownOptionProps = {
  optionNumber: number;
  dropdownText: string
}
```

and for the rest of the tests which require either of the optionNumber or dropdownText fields, define a simple Partial class like this:

```ts
async selectDropDownOption({ optionNumber, dropdownText }: Partial<DropdownOptionProps>){ ... }
```

At least then there's less type declaration going on and its all in one place

Reply:
agreed, i'll incorporate this suggestion